.github/workflows/test.yml

```diff
@@ -67,34 +72,35 @@ jobs:
         run: |
           pip cache remove llama_cpp_python
 
-      - name: Cache huggingface
-        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
-        with:
-          path: ~/.cache/huggingface
-          # config contains DEFAULT_MODEL
-          key: huggingface-${{ hashFiles('src/instructlab/configuration.py') }}
-
-      - name: Install instructlab and instructlab-eval
+      - name: Start inference server
         run: |
           export PATH="/home/runner/.local/bin:/usr/local/cuda/bin:$PATH"
-          python3 -m venv venv
-          . venv/bin/activate
           cd instructlab
+          python3.11 -m venv cli_venv
+          . cli_venv/bin/activate
           sed 's/\[.*\]//' requirements.txt > constraints.txt
-          python3 -m pip cache remove llama_cpp_python
+          python3.11 -m pip cache remove llama_cpp_python
           CMAKE_ARGS="-DLLAMA_CUBLAS=on" python3 -m pip install --no-binary llama_cpp_python -c constraints.txt llama_cpp_python
           # needed for --4-bit-quant option to ilab train
-          python3 -m pip install bitsandbytes
+          python3.11 -m pip install bitsandbytes
           # install instructlab
-          python3 -m pip install .
-          cd ..
-          # Install instructlab-eval
-          python3 -m pip install .
+          python3.11 -m pip install .
+          # start llama-cpp server
+          ilab model download --repository instructlab/granite-7b-lab-GGUF --filename granite-7b-lab-Q4_K_M.gguf
+          ilab model serve --model-path /home/runner/.local/share/instructlab/models/granite-7b-lab-Q4_K_M.gguf
```

Comment:
In the functional tests scripts we shut down the server to clean up. I wonder if we should split all of this into a shell script that manages ilab installation, server startup, pytest running, and server shutdown.

Reply:
Really, we shouldn't need `ilab` at all to run unit or functional tests for this library - it should operate independently of the CLI

Reply:
I'm gonna look into some other possible approaches

Reply:
> Really, we shouldn't need ilab at all to run unit or functional tests for this library - it should operate independently of the CLI

I think this makes sense for unit tests, but functional ones should prob still use a server from ilab etc